rnd/autogpt_server/autogpt_server/blocks/youtube_transcriber.py

```diff
@@ -0,0 +1,70 @@
+import re
+from urllib.parse import urlparse, parse_qs
+
+from youtube_transcript_api import YouTubeTranscriptApi
+from youtube_transcript_api.formatters import TextFormatter
+
+from autogpt_server.data.block import Block, BlockSchema, BlockOutput
+from autogpt_server.data.model import SchemaField
+
+class YouTubeTranscriber(Block):
+    class Input(BlockSchema):
+        youtube_url: str = SchemaField(
+            description="The URL of the YouTube video to transcribe",
+            placeholder="https://www.youtube.com/watch?v=dQw4w9WgXcQ"
+        )
+
+    class Output(BlockSchema):
+        video_id: str = SchemaField(description="The extracted YouTube video ID")
+        transcript: str = SchemaField(description="The transcribed text of the video")
+        error: str = SchemaField(description="Any error message if the transcription fails")
+
+    def __init__(self):
+        super().__init__(
+            id="f3a8f7e1-4b1d-4e5f-9f2a-7c3d5a2e6b4c",
+            input_schema=YouTubeTranscriber.Input,
+            output_schema=YouTubeTranscriber.Output,
+            test_input={
+                "youtube_url": "https://www.youtube.com/watch?v=dQw4w9WgXcQ"
+            },
+            test_output=[
+                ("video_id", "dQw4w9WgXcQ"),
+                ("transcript", "Never gonna give you up\nNever gonna let you down\n..."),
+            ],
+            test_mock={
+                "extract_video_id": lambda url: "dQw4w9WgXcQ",
```

Comment:
Hmmm `extract_video_id` should not be mocked, no ?
We need to test the parsing is correct

Reply:
Done